bionic/persistence.py

```diff
@@ -530,6 +530,10 @@ def list_items(self):
                 descriptor=metadata_record.descriptor,
             )
 
+    def delete_entry(self, metadata_url, artifact_url):
+        self._fs.delete(metadata_url)
```

Comment:
Let's delete the artifact before metadata in case there is some error while deleting the artifact url (like connection issues when deleting from gcs).

Reply:
The reason I delete the metadata first is because the metadata refers to the artifact; if you delete the artifact and then fail to delete the metadata, then the metadata will be invalid.